docker/oidc/mock-oidc-provider/start-server.sh

```diff
@@ -8,10 +8,36 @@ echo Waiting to make sure that oidc mock provider and proxy are running
 until $(curl --output /dev/null --silent --head --fail http://localhost:$OIDC_PROVIDER_PROXY_PORT/.well-known/openid-configuration); do
     sleep 0.3
 done
+
+echo Setting up user roles
+# Start the server (without auth).
+# This is original mongodb/mongodb-enterprise-server entrypoint
+python3 /usr/local/bin/docker-entrypoint.py \
+    --setParameter authenticationMechanisms="MONGODB-OIDC" \
+    --setParameter enableTestCommands="true" \
+    --setParameter oidcIdentityProviders="$OIDC_IDENTITY_PROVIDERS" > /dev/null &
+
+# Wait for the mongodb server to start.
+until nc -z localhost 27017; do
+    sleep 1
+done
+
+# Creates the OIDC user role in the database.
+mongosh "mongodb://localhost:27017/admin" --eval "JSON.stringify(db.createRole({ role: \"dev/groups\", privileges: [ ], roles: [ \"dbOwner\" ] }));"
```

Comment:
The role(s) would need to match the entries inside the `groups` claim, so that would be whatever `OIDC_TOKEN_PAYLOAD_GROUPS` specifies (and we probably want to use that because it is supposed to be a parameter for this image, right?)

```suggestion
mongosh "mongodb://localhost:27017/admin" --eval "JSON.stringify((process.env.OIDC_TOKEN_PAYLOAD_GROUPS ?? 'testgroup').split(',').map(group => db.createRole({ role: 'dev/' + group, privileges: [ ], roles: [ \"dbOwner\" ] })));"
```

or, if we’re sticking with a recent mongosh anyway,

```suggestion
mongosh "mongodb://localhost:27017/admin" --json --eval "(process.env.OIDC_TOKEN_PAYLOAD_GROUPS ?? 'testgroup').split(',').map(group => db.createRole({ role: 'dev/' + group, privileges: [ ], roles: [ \"dbOwner\" ] }));"
```

(otherwise, the default is `testgroup`)

Reply:
Tbh maybe I overkilled it by making them configurable, wouldn't mind if we don't want to do that, just felt like it wouldn't hurt 🙂

Reply:
I don’t think it’s a bad idea at all, but yeah, if we really want to leverage this, we’d probably also want to give the user the ability to apply a specific role to each group (i.e. make this a map group → role rather than a list of groups) so that behavior actually differs?

Reply:
Updated - thanks for the help on this.